wsd/DocumentBroker.cpp

```diff
@@ -615,6 +624,34 @@ void DocumentBroker::pollThread()
             }
             break;
 
+            case DocumentState::Activity::SyncFileTimestamp:
+            {
+                // Last upload failed, redo CheckFileInfo to reset the modified time.
+                assert(!isAsyncUploading() && "Unexpected async-upload in progress");
+
+#if !MOBILEAPP
+                if (!_checkFileInfo)
+                {
+                    const auto session = getFirstAuthorizedSession();
+                    if (!session)
+                    {
+                        // No session to synchronize the timestamp with.
+                        // Last resort; reset the timestamp and let it be.
+                        // We can't upload without a valid token anyway.
+                        LOG_WRN("No valid session to synchronize the timestamp with. Setting "
+                                "timestamp as unsafe");
+                        _storage->setLastModifiedTimeUnSafe();
```

Comment:
_storage is dereferenced here, but I see other places in this method where _storage is checked for null/empty. Should there be a check here for that? (or an assert to indicate that it cannot happen)